src/sentry/integrations/github/client.py

```diff
@@ -217,6 +217,13 @@ def get_commit(self, repo: str, sha: str) -> JSONData:
         commit: JSONData = self.get_cached(f"/repos/{repo}/commits/{sha}")
         return commit
 
+    def get_pullrequest_from_commit(self, repo: str, sha: str) -> JSONData:
+        """
+        https://docs.github.com/en/rest/commits/commits#list-pull-requests-associated-with-a-commit
+        """
+        pullrequest: JSONData = self.get(f"/repos/{repo}/commits/{sha}/pulls")
```

Comment:
It looks like this endpoint can return multiple pull requests ([source](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-pull-requests-associated-with-a-commit)). Are we handling this case here or will we error out?

Reply:
hmm the description says "Lists the merged pull request" singular

Reply:
Interesting API. Seems like it's returning one if commit is merged and multiple if it's not. In your case because the suspect commit is always merged it should be fine but I wouldn't fully trust it and put some checks in place to maybe fail silently if there are multiple?

Reply:
> If the commit is not present in the default branch, will only return open pull requests associated with the commit.

Sounds like the response will always be a list - so that should be uniform at least.
On our end - we'd only be fetching PR's from merged commits right? (suspect commit) So in that case we can just write a validation to ensure there's only ever a single PR in the list?

Reply:
nit: can we name this `pulls`?
that way it matches the api route as well